google/cloud/internal/async_polling_loop.cc

```diff
@@ -45,8 +45,12 @@ class AsyncPollingLoopImpl
   future<StatusOr<Operation>> Start(future<StatusOr<Operation>> op) {
     auto self = shared_from_this();
     auto w = WeakFromThis();
-    promise_ = promise<StatusOr<Operation>>([w] {
-      if (auto self = w.lock()) self->DoCancel();
+    auto const& options = CurrentOptions();
+    promise_ = promise<StatusOr<Operation>>([w, options] {
+      if (auto self = w.lock()) {
+        OptionsSpan span(std::move(options));
```

Comment:
The `operator()` is `const` in this lambda, the `std::move()` has no effect.  Do you want a mutable lambda?

Reply:
Good catch!

I see you disabled the "performance-move-const-arg" clang-tidy in #3458, which I approved, so I guess we just need to be careful.

Reply:
BTW.  I could be convinced that we should turn this back on.  @devjgm also has opinions on this

Reply:
SGTM